server/src/core/user/anonymous-service.ts

```diff
@@ -0,0 +1,48 @@
+import { 
+  User,
+  AnonymousService,
+  UserService,
+  AnonymousCreateArgs,
+  AnonymousDonateArgs
+} from './types';
+import { Transaction } from '../payment';
+import { rethrowIfAppError } from '../error';
+import { string } from '@hapi/joi';
+
+export interface AnonymousUsersArgs {
+  users: UserService
+};
+
+export class AnonymousUsers implements AnonymousService {
+  private users: UserService;
+
+  constructor(args: AnonymousUsersArgs) {
+    this.users = args.users;
+  }
+
+  async create(args: AnonymousCreateArgs): Promise<User> {
+    const { name, phone, email } = args;
+    try {
+      let user = await this.users.getByPhoneOrEmail(phone, email);
```

Comment:
I think getting by phone or email could be problematic. The problem is because we are not verifying the email. Let's say if person A is making an anonymous donation but then they decide to enter a phone number that does not exist in the system and person B's email. So in this case the user that will be returned is person B, and the transaction will be initiated for them.

Well I guess this would be a problem if they entered someone else's phone number. Of course the user will receive a prompt to accept or reject the transaction on their phone.

Maybe this is overthinking at this stage. For now we can just let this pass since we're still working in controlled environment, but I think we might need to think of ways to verify the information entered before proceeding with the transaction.

But somehow I feel calling `users.getByPhone` would be better than `users.getByPhoneOrEmail` since at least the phone number is verified by the fact the they get a prompt on their phone, but the email is not verified during this process.